oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/inventory/IndexPrinter.java

```diff
@@ -66,97 +68,190 @@ public IndexPrinter(IndexInfoService indexInfoService, AsyncIndexInfoService asy
-            pw.println();
+            printTitle(pw, title, format);
+            addJsonKey(json, type);
+            startJsonObject(json);
             for (IndexInfo info : typedInfo){
-                printIndexInfo(pw, info);
+                printIndexInfo(pw, json, info, format);
+                if (format == Format.TEXT) {
+                    if (info.hasIndexDefinitionChangedWithoutReindexing()) {
+                        pw.println("    Index definition changed without reindexing");
+                        String diff = info.getIndexDefinitionDiff();
+                        if (diff != null) {
+                            pw.println("    "+diff);
+                        }
+                    }
+                    pw.println();
+                }
             }
+            endJsonObject(json);
         }
     }
 
-    private static void printIndexInfo(PrintWriter pw, IndexInfo info) {
-        pw.println(info.getIndexPath());
-        pw.printf("    Type                    : %s%n", info.getType());
+
+    private static void printIndexInfo(PrintWriter pw, JsopBuilder json, IndexInfo info, Format format) {
+        if (format == Format.TEXT) {
+            pw.println(info.getIndexPath());
+        }
+        addJsonKey(json, info.getIndexPath());
+        startJsonObject(json);
+
+        keyValue("    Type                    ", info.getType(), pw, json, format);
         if (info.getAsyncLaneName() != null) {
-            pw.printf("    Async                   : true%n");
-            pw.printf("    Async lane name         : %s%n", info.getAsyncLaneName());
+            keyValue("    Async                    ", true, pw, json, format);
+            keyValue("    Async lane name          ", info.getAsyncLaneName(), pw, json, format);
         }
 
         if (info.getIndexedUpToTime() > 0){
-            pw.printf("    Last indexed up to      : %s%n", formatTime(info.getIndexedUpToTime()));
+            keyValue("    Last indexed up to       ", formatTime(info.getIndexedUpToTime()), pw, json, format);
         }
 
         if (info.getLastUpdatedTime() > 0){
-            pw.printf("    Last updated time       : %s%n", formatTime(info.getLastUpdatedTime()));
+            keyValue("     Last updated time       ", formatTime(info.getLastUpdatedTime()), pw, json, format);
+        }
+
+        if (info.getCreationTimestamp() > 0){
+            keyValue("     Creation time           ", formatTime(info.getCreationTimestamp()), pw, json, format);
+        }
+
+        if (info.getReindexCompletionTimestamp() > 0){
+            keyValue("     Reindex completion time ", formatTime(info.getReindexCompletionTimestamp()), pw, json, format);
         }
 
         if (info.getSizeInBytes() >= 0){
-            pw.printf("    Size                    : %s%n", IOUtils.humanReadableByteCount(info.getSizeInBytes()));
+            keyValue("    Size                     ", IOUtils.humanReadableByteCount(info.getSizeInBytes()), pw, json, format);
```

Comment:
My personal opinion is that we should list all size data in the same way, e.g. all in GB. Examples:
* 1000 (= 1 TB)
* 1 (= 1 GB)
* 0.001 (= 1 MB)
* 0.000001 (= 1 B)

Reply:
@thomasmueller  - I disagree a bit here. Since this is human readable format (that would not be consumed by any code for calculations/alerting ), so I think, 1 B makes more sense than 0.000001 GB. 

If we want to use these for some calculations/alerting - then I agree with your point that they should be in the same unit (and in which case we would have to change the formatting even a bit more ), because here the o/p would be string - 10B, 10MB, 10GB and like that.

Reply:
> I disagree a bit here.

That's fine, it's just my personal opinion. When I investigate a problem, in the Lucene index stats, it I usually search for "GB" and then look at only those that have this unit; or use grep "GB". I would miss entries with "TB" however. 

So I argue that even for a human, using the same unit would make sense, assuming there is a sensible default unit.

We could round to 3 digits after the comma (e.g. round up from 0.000001 GB => 0.001 GB).

Reply:
> If we want to use these for some calculations/alerting

Looking at the json output, I think it would make sense to use it for alerting... What about we list both the "human readable" as well as the "same unit" numbers?

Reply:
Makes sense. I will add both the String format (that appends  B, MB, GB  accordingly) and the numeric one (in bytes).